tests/std/tests/P0896R4_views_common/test.cpp

```diff
@@ -0,0 +1,386 @@
+
+            // Validate common_view::size
+            static_assert(CanMemberSize<R> == sized_range<Rng>);
+            if constexpr (sized_range<Rng>) {
+                assert(r.size() == static_cast<range_size_t<R>>(size(expected)));
+                // static_assert(noexcept(r.size()) == noexcept(size(rng)));
+            }
+
+            static_assert(CanMemberSize<const R> == sized_range<const Rng>);
+            if constexpr (sized_range<const Rng>) {
+                assert(as_const(r).size() == static_cast<range_size_t<R>>(size(expected)));
+                // static_assert(noexcept(r.size()) == noexcept(size(as_const(rng))));
+            }
+
+            // Validate view_interface::empty and operator bool
+            const bool is_empty = ranges::empty(expected);
+            if (!is_constant_evaluated()) {
+                if constexpr (CanMemberEmpty<R>) {
+                    assert(r.empty() == is_empty);
+                    assert(static_cast<bool>(r) == !is_empty);
+                }
+                if constexpr (CanMemberEmpty<const R>) {
+                    assert(as_const(r).empty() == is_empty);
+                    assert(static_cast<bool>(as_const(r)) == !is_empty);
+                }
+            }
+
+            // Validate common_view::begin
+            STATIC_ASSERT(CanMemberBegin<R>);
+            STATIC_ASSERT(CanBegin<const R&> == range<const V>);
+            {
+                const same_as<iterator_t<R>> auto i = r.begin();
+                if (!is_empty) {
+                    assert(*i == *begin(expected));
+                }
+
+                if constexpr (copyable<V>) {
+                    auto r2                              = r;
+                    const same_as<iterator_t<R>> auto i2 = r2.begin();
+                    if (!is_empty) {
+                        assert(*i2 == *i);
+                    }
+                }
+
+                if constexpr (CanBegin<const R&>) {
+                    const same_as<iterator_t<const R>> auto i3 = as_const(r).begin();
+                    if (!is_empty) {
+                        assert(*i3 == *i);
+                    }
+                }
+            }
+            // Validate common_view::end
+            STATIC_ASSERT(CanMemberEnd<R>);
+            STATIC_ASSERT(CanEnd<const R&> == range<const V>);
+            if (!is_empty) {
+                same_as<iterator_t<R>> auto i = r.end();
+                if constexpr (bidirectional_range<R>) {
+                    assert(*prev(i) == *prev(end(expected)));
+                }
+
+                if constexpr (CanEnd<const R&>) {
+                    same_as<iterator_t<const R>> auto i2 = as_const(r).end();
+                    if constexpr (bidirectional_range<const R>) {
+                        assert(*prev(i2) == *prev(end(expected)));
+                    }
+                }
+            }
+
+            // Validate view_interface::data
+            static_assert(!CanData<R>);

```

Comment:
I find this super troubling. If the user passes a `continuous_range` into `common_view` he will be in for a surprise

Reply:
This is a backwards-compatibility tool to adapt non-`common_range`s into an iterator pair that can be passed to C++17 algorithms. It doesn't need to have great support for `contiguous_range`s that don't exist in C++17.